src/keywrap/jwe.rs

```diff
@@ -0,0 +1,230 @@
+use crate::keywrap::KeyWrapper;
+use josekit::jwk::{Jwk, KeyAlg, KeyFormat, KeyInfo};
+
+use josekit::jwe::{
+    deserialize_json, serialize_general_json, JweDecrypter, JweEncrypter, JweHeader, JweHeaderSet,
+    ECDH_ES_A256KW, RSA_OAEP,
+};
+
+/// A Jwe keywrapper
+#[derive(Debug)]
+pub struct JweKeyWrapper {}
+
+// Get the encrypter from public key
+fn get_encrypter(pubkey: &[u8]) -> Result<Box<dyn JweEncrypter>> {
+    let key_info = match KeyInfo::detect(&pubkey) {
+        Some(i) => i,
+        None => return Err(anyhow!("failed to detect public key info")),
+    };
+
+    if !key_info.is_public_key() {
+        return Err(anyhow!("expect public key, found private key"));
+    }
+
+    let alg = match key_info.alg().unwrap() {
+        KeyAlg::Ec { curve: Some(_) } => "ECDH_ES_A256KW",
+        _ => "RSA_OAEP",
+    };
+
+    if alg == "RSA_OAEP" {
+        let rsa_encrypter = match key_info.format() {
+            KeyFormat::Der { raw: _ } => RSA_OAEP.encrypter_from_der(pubkey)?,
+            KeyFormat::Pem { traditional: _ } => RSA_OAEP.encrypter_from_pem(pubkey)?,
+            KeyFormat::Jwk => RSA_OAEP.encrypter_from_jwk(&Jwk::from_bytes(pubkey)?)?,
+        };
+
+        Ok(rsa_encrypter.box_clone())
+    } else {
+        let ecdh_encrypter = match key_info.format() {
+            KeyFormat::Der { raw: _ } => ECDH_ES_A256KW.encrypter_from_der(pubkey)?,
+            KeyFormat::Pem { traditional: _ } => ECDH_ES_A256KW.encrypter_from_pem(pubkey)?,
+            KeyFormat::Jwk => ECDH_ES_A256KW.encrypter_from_jwk(&Jwk::from_bytes(pubkey)?)?,
+        };
+
+        Ok(ecdh_encrypter.box_clone())
+    }
+}
+
+// Get the decrypter from private key
+fn get_decrypter(priv_key: &[u8]) -> Result<Box<dyn JweDecrypter>> {
+    let key_info = match KeyInfo::detect(&priv_key) {
+        Some(i) => i,
+        None => return Err(anyhow!("failed to detect private key info")),
+    };
+
+    if key_info.is_public_key() {
+        return Err(anyhow!("expect private key, found public key"));
+    }
+
+    let alg = match key_info.alg().unwrap() {
+        KeyAlg::Ec { curve: Some(_) } => "ECDH_ES_A256KW",
+        _ => "RSA_OAEP",
+    };
+
+    if alg == "RSA_OAEP" {
+        let rsa_decrypter = match key_info.format() {
+            KeyFormat::Der { raw: _ } => RSA_OAEP.decrypter_from_der(priv_key)?,
+            KeyFormat::Pem { traditional: _ } => RSA_OAEP.decrypter_from_pem(priv_key)?,
+            KeyFormat::Jwk => RSA_OAEP.decrypter_from_jwk(&Jwk::from_bytes(priv_key)?)?,
+        };
+
+        Ok(rsa_decrypter.box_clone())
+    } else {
+        let ecdh_decrypter = match key_info.format() {
+            KeyFormat::Der { raw: _ } => ECDH_ES_A256KW.decrypter_from_der(priv_key)?,
+            KeyFormat::Pem { traditional: _ } => ECDH_ES_A256KW.decrypter_from_pem(priv_key)?,
+            KeyFormat::Jwk => ECDH_ES_A256KW.decrypter_from_jwk(&Jwk::from_bytes(priv_key)?)?,
+        };
+
+        Ok(ecdh_decrypter.box_clone())
+    }
+}
```

Comment:
I have very similar comments for this one than for the `get_encrypter` function.